internal/proxy/domain/model/node_type.go

```diff
@@ -0,0 +1,11 @@
+package model
+
+// NodeType represents the int
+type NodeType int
+
+// ArchiveNode and RollingNode node types
+const (
+    ArchiveNode NodeType = iota + 1
```

Comment:
The syntax is a bit strange... Is it normal ?

Reply:
Yes, it's Go's syntax for enum

Reply:
Ok ugly but good enough. I approve the PR